src/mergeAxeResults.ts

```diff
@@ -451,15 +465,197 @@ async function compressJsonFileStreaming(inputPath: string, outputPath: string)
   console.log(`File successfully compressed and saved to ${outputPath}`);
 }
 
+const appendJsonChunksToFile = async (chunks, filePath) => {
```

Comment:
This seems to be doing some of what `serializeObject` is doing without using `serializeObject`, and only applying it on the smaller objects inside. It avoids the error by reducing the depth of the recursive calls.

I've written an iterative version of `serializeObject` in [this commit](https://github.com/GovTechSG/oobee/pull/425/commits/e7fe180f2d6df9a8fc98baf9139a7de87e24cbfc).